src/encoded/static/components/genome_browser.js

```diff
@@ -703,23 +703,28 @@ class GenomeBrowser extends React.Component {
     /* eslint-disable react/no-did-update-set-state */
     componentDidUpdate(prevProps, prevState) {
         if (!(this.state.disableBrowserForIE) && this.GV) {
-            if (this.state.contig !== prevState.contig) {
+            const contigUpdate = this.state.contig !== prevState.contig;
+            const assemblyUpdate = this.props.assembly !== prevProps.assembly || this.props.annotation !== prevProps.annotation;
+            const filesUpdate = !(_.isEqual(this.props.files, prevProps.files));
```

Comment:
_.isEqual does a deep copy. Is that needed as a shallow copy is faster if it is possible.

Reply:
I think so, though I wrote this code a while ago.